test/reservation_desk_test.rb

```diff
@@ -0,0 +1,317 @@
+      expect(result_2.length).must_equal 5
+    end
+
+    it "No ID provided: returns empty array if there are no reservations for the room/dates" do
+      result = @reservation_desk.find_reservations(room_id: 1, start_date: "2020-4-1", end_date: "2020-4-17")
+      expect(result.length).must_equal 0
+    end
+  end
+
+  describe "find_available_rooms" do
+    before do
+      @reservation_desk.rooms[0].reserve(start_date: "2020-3-1", end_date: "2020-3-5")
+      @reservation_desk.rooms[1].reserve(start_date: "2020-3-1", end_date: "2020-3-10")
+      @reservation_desk.rooms[2].reserve(start_date: "2020-3-1", end_date: "2020-3-25")
+
+      @request = @reservation_desk.find_available_rooms(start_date: "2020-2-27", end_date: "2020-4-2")
+    end
+
+    it "returns an Array" do
+      expect(@request).must_be_kind_of Array
+    end
+
+    it "returns an Array of Rooms" do
+      expect(@request[0]).must_be_kind_of Hotel::Room
+    end
+
+    it "returns an array of Rooms which don't have Reservations for requested dates" do
+      expect(@request.length).must_equal 17
+      expect(@request[0].id).must_equal 4
+    end
+
+    it "returns nil if there are no Rooms available for requested dates" do
+      @reservation_desk.rooms.each do |room|
+        room.reserve(start_date: "2020-6-1", end_date: "2020-6-30")
+      end
+      search = @reservation_desk.find_available_rooms(start_date: "2020-6-5", end_date: "2020-6-10")
+      assert_nil search
+    end
+  end
+
+  describe "make_reservation" do
+    it "ID provided: makes a new reservation and add it into the Room's reservations array" do
+      @reservation_desk.make_reservation(room_id: 5, start_date: "2020-5-5", end_date: "2020-5-7")
+      @reservation_desk.make_reservation(room_id: 5, start_date: "2020-6-5", end_date: "2020-6-7")
+      expect(@reservation_desk.rooms[4].reservations.length).must_equal 2
+      expect(@reservation_desk.rooms[4].available?(start_date: "2020-5-5", end_date: "2020-5-6")).must_equal false
+    end
+
+    it "ID provided: Raises an ArgumentError if ID is invalid" do
+      expect {
+        @reservation_desk.make_reservation(room_id: 999999, start_date: "2020-5-5", end_date: "2020-5-7")
+      }.must_raise ArgumentError
+    end
+
+    it "No ID provided: reserves the first available room down the list for requested dates" do
+      @reservation_desk.rooms[0].reserve(start_date: "2020-3-1", end_date: "2020-3-5")
+      @reservation_desk.rooms[1].reserve(start_date: "2020-3-1", end_date: "2020-3-10")
+
+      @request = @reservation_desk.make_reservation(start_date: "2020-3-4", end_date: "2020-3-15")
+      expect(@reservation_desk.rooms[2].reservations.length).must_equal 1
+      expect(@reservation_desk.rooms[2].available?(start_date: "2020-3-1", end_date: "2020-3-5")).must_equal false
+    end
+
+    it "No ID provided: raises an Exception if no rooms are available for requested dates" do
```

Comment:
Great test of an edge case. Minor note: Consider making a custom exception for this situation.